src/components/network-map-viewer/network/network-map.jsx

```diff
@@ -658,6 +797,30 @@ NetworkMap.propTypes = {
     onSubstationClickChooseVoltageLevel: PropTypes.func,
     onSubstationMenuClick: PropTypes.func,
     onVoltageLevelMenuClick: PropTypes.func,
+    onDrawModeChanged: PropTypes.func,
+    onFeaturesChanged: PropTypes.func,
 };
 
 export default React.memo(NetworkMap);
+
+function getSubstationsInPolygon(features, mapEquipments, geoData) {
+    const firstPolygonFeatures = Object.values(features)[0];
+    const polygonCoordinates = firstPolygonFeatures?.geometry;
+    if (!polygonCoordinates || polygonCoordinates.coordinates < 3) {
+        return [];
+    }
+    //get the list of substation
+    const substationsList = mapEquipments?.substations ?? [];
+    //for each substation, check if it is in the polygon
+    return substationsList // keep only the sybstation in the polygon
+        .filter((substation) => {
+            const pos = geoData.getSubstationPosition(substation.id);
+            return booleanPointInPolygon(pos, polygonCoordinates);
+        });
+}
+
+function getVoltageLevelFromSubstation(substations) {
+    return substations.flatMap((substation) => {
+        return substation.voltageLevels;
+    });
```

Comment:
I think we want distinct VoltageLevels here no ?
```suggestion
    return [...new Set(substations.flatMap((substation) => substation.voltageLevels))];
```